0004-ledger-plugin-interface/0004-ledger-plugin-interface.md

```diff
@@ -226,25 +231,25 @@ p.sendTransfer({
 })
 ```
 
-For a detailed description of these properties, please see [`OutgoingTransfer`](#outgoingtransfer).
+For a detailed description of these properties, please see [`Class: Transfer`](#class-transfer).
 
-#### sendMessage
-<code>ledgerPlugin.sendMessage( **message**:[OutgoingMessage](#outgoingmessage) ) ⇒ Promise.&lt;null></code>
+#### sendRequest
+<code>ledgerPlugin.sendRequest( **message**:[Message](#class-message) ) ⇒ Promise.&lt;[Message](#class-message)></code>
 
-Plugin must be connected, otherwise the promise should reject. Send a ledger-local message.
-If there is a problem with the structure or validity of the message, then `sendMessage` should reject with an error.
+Plugin must be connected, otherwise the promise should reject. Sends a ledger-local message. Returns a promise for a response message from the other side.
+If there is a problem with the structure or validity of the message, then `sendRequest` should reject with an error.
 
 Messaging is used by connectors for [quoting](../0008-interledger-quoting-protocol/) and broadcasting routes.
 
 ###### Parameters
 | Name | Type | Description |
 |:--|:--|:--|
-| message | <code>[OutgoingMessage](#outgoingmessage)</code> | Properties of the message to be created |
+| message | <code>[Message](#class-message)</code> | Properties of the message to be created |
 
-When sending messages, the [to](#to) and [data](#data) fields are required.
+When sending messages, the [to](#message-to) and [ilp](#message-ilp) fields are required.
 
 ###### Returns
-**`Promise.<null>`** A promise which resolves when the message has been submitted (but not necessarily delivered). To ensure delivery, you must build a mechanism for that on top.
+**<code>Promise.<[Message](#class-message)></code>** A promise which resolves when a response message has been received.
```

Comment:
> Is the expectation that protocols built on top of this will define more status codes, including something like an HTTP 200, or will those be included in the standard set of ILP errors?

Not sure I understand the question correctly, but if there is an error in the ledger protocol (e.g. an HTTP 500 for an HTTP based protocol), the plugin should probably still generate a standard ILP error and return that instead.

> Will errors also be communicated as ILP packets?

Yes, errors will always be in the ILP error format as described in the ASN.